Packages/vcs/Lib/vcsvtk/isofillpipeline.py

```diff
@@ -0,0 +1,296 @@
+
+        for i, l in enumerate(self._contourLevels):
+            if i == 0:
+                C = [self._contourColors[i],]
+                if numpy.allclose(self._contourLevels[0][0], -1.e20):
+                    ## ok it's an extension arrow
+                    L = [self._scalarRange[0] - 1., self._contourLevels[0][1]]
+                else:
+                    L = list(self._contourLevels[i])
+                I = [indices[i],]
+            else:
+                if l[0] == L[-1] and I[-1] == indices[i]:
+                    # Ok same type lets keep going
+                    if numpy.allclose(l[1], 1.e20):
+                        L.append(self._scalarRange[1] + 1.)
+                    else:
+                        L.append(l[1])
+                    C.append(self._contourColors[i])
+                else: # ok we need new contouring
+                    tmpLevels.append(L)
+                    tmpColors.append(C)
+                    C = [self._contourColors[i],]
+                    L = tmpLevels[i]
+                    I = [indices[i],]
+        tmpLevels.append(L)
+        tmpColors.append(C)
+
+        luts = []
+        cots = []
+        geos = []
+        mappers = []
+        for i,l in enumerate(tmpLevels):
+            # Ok here we are trying to group together levels can be, a join will
+            # happen if: next set of levels contnues where one left off AND
+            # pattern is identical
+            mapper = vtk.vtkPolyDataMapper()
+            lut = vtk.vtkLookupTable()
+            cot = vtk.vtkBandedPolyDataContourFilter()
+            cot.ClippingOn()
+            cot.SetInputData(self._vtkPolyDataFilter.GetOutput())
+            cot.SetNumberOfContours(len(l))
+            cot.SetClipTolerance(0.)
+            for j,v in enumerate(l):
+               cot.SetValue(j,v)
+            cot.Update()
+            cots.append(cot)
+            mapper.SetInputConnection(cot.GetOutputPort())
+            lut.SetNumberOfTableValues(len(tmpColors[i]))
+            for j, color in enumerate(tmpColors[i]):
+                r, g, b = self._colorMap.index[color]
+                lut.SetTableValue(j, r / 100., g / 100., b / 100.)
+            luts.append([lut, [0, len(l) - 1, True]])
+            mapper.SetLookupTable(lut)
+            mapper.SetScalarRange(0, len(l) - 1)
+            mapper.SetScalarModeToUseCellData()
+            mappers.append([mapper,])
+
+        self._resultDict["vtk_backend_luts"] = luts
+        if len(cots) > 0:
+            self._resultDict["vtk_backend_contours"] = cots
+        if len(geos) > 0:
+            self._resultDict["vtk_backend_geofilters"] = geos
+
+        numLevels = len(self._contourLevels)
+        if mappers == []: # ok didn't need to have special banded contours
+            # TODO Charles, this branch will never execute, unless numLevels==0.
+            # Is that a supported usecase, or should that throw an error?
+            mapper = vtk.vtkPolyDataMapper()
```

Comment:
```
levels =  ([1.0000000200408773e+20, 1.0000000200408773e+20],)
```
is the closest thing from level-less it means "I trust you computer, give me your best shot"